lib/Transforms/IPO/PassManagerBuilder.cpp

```diff
@@ -501,7 +501,10 @@ void PassManagerBuilder::populateModulePassManager(
     }
     // HLSL Change Ends
   }
+
   // HLSL Change Begins.
+  // Use value numbering to figure out if regions are equivalent, and branch to only one.
+  MPM.add(createDxilSimpleGVNEliminateRegionPass());
```

Comment:
If this is redundant with GVN, should we only do it when GVN is disabled?

Reply:
Done.

Reply:
I think it's fine to run them both, there's not much downsides to running this, it might catch some redundant code.